core/src/main/java/org/elasticsearch/action/bulk/BulkItemRequest.java

```diff
@@ -94,15 +94,36 @@ public void readFrom(StreamInput in) throws IOException {
         request = DocWriteRequest.readDocumentRequest(in);
         if (in.readBoolean()) {
             primaryResponse = BulkItemResponse.readBulkItem(in);
+            if (in.getVersion().onOrAfter(Version.V_5_6_0_UNRELEASED)) {
```

Comment:
I am not sure we should mix `onOrAfter` and `before` in the same file, it's confusing?